### DISCUSSION TITLE
Match more file regex rules with prefix or suffix

### DISCUSSION THREAD

**Discussion by jsonchou** [vercel/next.js#21071] [Help]
Hi guys, 

It's there any way to match more file regex rules with prefix and suffix 

For now, i can use, 
``` js
/pages/foo/tag/[:id].tsx -> pages/foo/tag/123
/pages/foo/detail/[:id].tsx -> pages/foo/detail/123
/pages/foo/special/[:id].tsx -> pages/foo/special/123
/pages/foo/question/[:id].tsx -> pages/foo/question/123
```

But, how can i make the path more and more shorter like this? 
``` js
/pages/foo/t[:id].tsx -> pages/foo/t123 
/pages/foo/d[:id].tsx -> pages/foo/d123 
/pages/foo/s[:id].tsx -> pages/foo/s123
/pages/foo/q[:id].tsx -> pages/foo/q123
```

**Comment by aiji42** [comment]
I too would like to know the best practices on this topic. Or, I'm eagerly waiting for the feature improvement so that paths with prefixes and suffixes can be resolved flexibly in dynamic routing.

On our site, I have to interpret the following url.(We are well aware that we should review the design of the url in the first place, but we are taking a cautious approach because we are adapting to an existing site.)
For example, suppose we have a site that allows we to search for property information with a URL hierarchy of `prefecture > city > specific conditions`. The city and specific conditions can be omitted, which implicitly indicates that the path is "all".
```
/pref-tokyo/city-123/condition-xxxxx/list
/pref-kyoto/city-456/condition-yyyyy/list
/pref-kyoto/list
/pref-osaka/city-789/list
/pref-osaka/condition-yyyyy/list
```
The number of hierarchies will fluctuate in this way. Also, since the content displayed for each pattern is slightly different, we want to keep the files as separate as possible.
Currently, the second and third paths have to be merged into a single file layout. It is also necessary to remove the prefix from the parameters, which makes the process more complicated.
```
/pages/[prefecture]/list.tsx
/pages/[prefecture]/[cityOrCondition]/list.tsx
/pages/[prefecture]/[city]/[condition]/list.tsx
```

---
Thus, if dynamic routing is supported for prefixed filenames, it can provide a manageable and flexible file layout.
```
/pages/pref-[prefecture]/list.tsx
/pages/pref-[prefecture]/city-[city]/list.tsx
/pages/pref-[prefecture]/condition-[condition]/list.tsx
/pages/pref-[prefecture]/city-[city]/condition-[coundition]/list.tsx
```
